packages/pigeon/platform_tests/test_plugin/ios/Classes/CoreTests.gen.swift

```diff
@@ -652,6 +652,66 @@ class CoreTestsPigeonCodec: FlutterStandardMessageCodec, @unchecked Sendable {
   static let shared = CoreTestsPigeonCodec(readerWriter: CoreTestsPigeonCodecReaderWriter())
 }
 
+private class PigeonStreamHandler<T>: NSObject, FlutterStreamHandler {
+  private let wrapper: PigeonEventChannelWrapper<T>
+  private var pigeonSink: PigeonEventSink<T>? = nil
+
+  init(wrapper: PigeonEventChannelWrapper<T>) {
+    self.wrapper = wrapper
+  }
+
+  func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink)
+    -> FlutterError?
+  {
+    pigeonSink = PigeonEventSink<T>(events)
+    wrapper.onListen(withArguments: arguments, sink: pigeonSink!)
+    return nil
+  }
+
+  func onCancel(withArguments arguments: Any?) -> FlutterError? {
+    pigeonSink = nil
+    wrapper.onCancel(withArguments: arguments)
+    return nil
+  }
+}
+
+class PigeonEventChannelWrapper<T> {
+  func onListen(withArguments arguments: Any?, sink: PigeonEventSink<T>) {}
+  func onCancel(withArguments arguments: Any?) {}
+}
+
+class PigeonEventSink<T> {
```

Comment:
It looks like this is just a 1-line wrapper? Can we just inline the calls?

Reply:
What do you mean by "inline" specifically?

Reply:
Can we directly use the wrapped type rather than using this wrapper?

Reply:
no, the purpose of this wrapper is to enforce type safety on the return value

Reply:
By type safety you mean the generic type `T` in the `success(_ value:)` call?

Reply:
yeah, so I see what you mean, the error method isn't needed. I'm not sure if this is it's final form, but if it is, I'll remove it.